client/client.go

```diff
@@ -1597,10 +1437,34 @@ func (c *client) respForErr(observer prometheus.Observer, start time.Time, err e
 	if err != nil || header.GetError() != nil {
 		observer.Observe(time.Since(start).Seconds())
 		if err != nil {
-			c.pdSvcDiscovery.ScheduleCheckMemberChanged()
+			c.inner.pdSvcDiscovery.ScheduleCheckMemberChanged()
 			return errors.WithStack(err)
 		}
 		return errors.WithStack(errors.New(header.GetError().String()))
 	}
 	return nil
 }
+
+// WithCallerComponent implements the RPCClient interface.
+func (c *client) WithCallerComponent(callerComponent caller.Component) RPCClient {
+	newClient := *c
+	newClient.callerComponent = callerComponent
+	return &newClient
+}
+
+// adjustCallerComponent returns the caller component if it is empty, it
+// is the upper layer of the pd client.
+func adjustCallerComponent(callerComponent caller.Component) caller.Component {
+	if len(callerComponent) != 0 {
```

Comment:
Better trim it first before the check.

Reply:
Trim space? https://github.com/tikv/pd/pull/8704/commits/de83c86db37420405deea689e68894c9d39fc2be